javascript/javascript3/week1/weatherApp/weather.js

```diff
@@ -0,0 +1,144 @@
+//Assigning api key to the variable
+const apiKey = '2a984eff06dbace9308d2bcf60b5e917';
+let weatherApi;
+
+//function that gets the html elements by id
+function getHtmlMarkup(idName) {
+    return document.getElementById(idName);
+}
+
+
+
+//
+const options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+let todayDate;
+const todaysDate =  getHtmlMarkup('todaysDateH2');
+const currentTime = getHtmlMarkup('currentTimeH2');
+const currentDetailsUl = getHtmlMarkup('currentDetailsUlMarkup');
+//below functions trigger when the page is loaded
+document.addEventListener('DOMContentLoaded', onLoad);
+
+//call the function to get the current date and time in your location and get the last city searched from the local storage
+function onLoad() {
+    getcurrentDetails();
+    const cityName = localStorage.getItem('cityName');
+    if(cityName) {
+       getWeatherByCity(cityName);
+    }
+}
+// function to get the current date and time in your location
+function getcurrentDetails () {
+    todayDate = new Date();
+    todaysDate.innerHTML = todayDate.toLocaleDateString(undefined, options);
+    currentTime.innerHTML = todayDate.toLocaleTimeString('en-US');
+}
+setInterval(getcurrentDetails, 1000); //to update time every second
+
+
+//get user entered city location and set the cityname in the API link when get the city location weather button clicked
+const weatherButton = getHtmlMarkup('weatherButtonMarkup');
+
+weatherButton.addEventListener('click', () => {
+    const cityInput = getHtmlMarkup('cityInputMarkup').value.trim();
+    (cityInput)? getWeatherByCity(cityInput) : alert('Please Enter The City Name');;
+});
+
+function getWeatherByCity(city) {
+    weatherApi = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`;
+    displayWeatherData(weatherApi);
+}
+
+
+//get the current location coordinates and set the coordinates in the API link when get the current location weather button clicked
+const currentlocationweather = getHtmlMarkup('currentlocationweatherButton');
+let locationLatitude;
+let locationLongitude;
+
+currentlocationweather.addEventListener('click', function() {
+    navigator.geolocation.getCurrentPosition(success,error);
+});
+
+function success(pos) {
+    var crd = pos.coords;
```

Comment:
**Please don't use 'var'**
In the Es6 version of JavaScript, variables can be declared in two more ways. The first is **'let'** and the second **'const'.**